hack/functests.sh

```diff
@@ -63,45 +63,18 @@ function functest() {
     _out/tests/ginkgo -timeout=3h -r -slow-spec-threshold=60s $@ _out/tests/tests.test -- -kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra} -config=${kubevirt_test_config} --artifacts=${ARTIFACTS} --operator-manifest-path=${OPERATOR_MANIFEST_PATH} ${KUBEVIRT_FUNC_TEST_SUITE_ARGS}
 }
 
-if [ "$KUBEVIRT_E2E_PARALLEL" == "true" ]; then
-    trap "_out/tests/junit-merger -o ${ARTIFACTS}/junit.functest.xml '${ARTIFACTS}/partial.*.xml'" EXIT
-    parallel_test_args=""
-    serial_test_args=""
-
-    if [ -n "$KUBEVIRT_E2E_SKIP" ]; then
-        parallel_test_args="${parallel_test_args} --skip=\\[Serial\\]|${KUBEVIRT_E2E_SKIP}"
-        serial_test_args="${serial_test_args} --skip=${KUBEVIRT_E2E_SKIP}"
-    else
-        parallel_test_args="${parallel_test_args} --skip=\\[Serial\\]"
-    fi
+additional_test_args=""
+if [ -n "$KUBEVIRT_E2E_SKIP" ]; then
+    additional_test_args="${additional_test_args} --skip=${KUBEVIRT_E2E_SKIP}"
+fi
 
-    if [ -n "$KUBEVIRT_E2E_FOCUS" ]; then
-        parallel_test_args="${parallel_test_args} --focus=${KUBEVIRT_E2E_FOCUS}"
-        serial_test_args="${serial_test_args} --focus=\\[Serial\\].*(${KUBEVIRT_E2E_FOCUS})|(${KUBEVIRT_E2E_FOCUS}).*\\[Serial\\]"
-    else
-        serial_test_args="${serial_test_args} --focus=\\[Serial\\]"
-    fi
+if [ -n "$KUBEVIRT_E2E_FOCUS" ]; then
+    additional_test_args="${additional_test_args} --focus=${KUBEVIRT_E2E_FOCUS}"
+fi
 
-    return_value=0
-    set +e
-    functest --nodes=${KUBEVIRT_E2E_PARALLEL_NODES} ${parallel_test_args} ${KUBEVIRT_FUNC_TEST_GINKGO_ARGS}
-    return_value="$?"
-    set -e
-    if [ "$return_value" -ne 0 ] && ! [ "$KUBEVIRT_E2E_RUN_ALL_SUITES" == "true" ]; then
-        exit "$return_value"
-    fi
-    KUBEVIRT_FUNC_TEST_SUITE_ARGS="-junit-output ${ARTIFACTS}/partial.junit.functest.xml ${KUBEVIRT_FUNC_TEST_SUITE_ARGS}"
-    functest ${serial_test_args} ${KUBEVIRT_FUNC_TEST_GINKGO_ARGS}
-    exit "$return_value"
+if [ "$KUBEVIRT_E2E_PARALLEL" == "true" ]; then
+    trap "_out/tests/junit-merger -o ${ARTIFACTS}/junit.functest.xml '${ARTIFACTS}/partial.*.xml'" EXIT
```

Comment:
As there is only one execution (and not two as before), is there still a need to perform the merges? Are there indeed individual files for each parallel run?
(I thought there are two: One for the parallel run and another for the serial one)

This was first raised [here](https://github.com/kubevirt/kubevirt/pull/7620#issuecomment-1107859634).

Reply:
I think this is what I see in the artifacts as well [1].
There are always two "partial" junit files and the 3rd is the merge result.

On this PR result, I can see only a single one [2].

[1] https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7627/pull-kubevirt-e2e-k8s-1.23-sig-network/1518464819684970496/artifacts/

[2] https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7620/pull-kubevirt-e2e-k8s-1.21-sig-network/1518208895724556288/artifacts/

Reply:
lets remove it and ask them to update their scripts ?
junit-merger is doing some conversion as well, so it is better to remove code (however it will still kept on unit-tests which need more work in order to remove it)
(same for the previous "[Serial]")

if you use junit-report and run the trap it will fail, because there will be duplicate sync sections
removing the trap will be a good preparation for adapting the ginkgo 2.0 built-in --junit report

Reply:
> Unfortunately, we still need it, because other clients use these tests, and expect the partial.*.xml file name format. That can be easily fixed, using `--junit-report` parameter when running the tests.

What clients? I am unclear on why someone will depend on the intermediate artifacts.
I guess I am missing some context here, but my understanding that those files are just "partial" temporary artifacts from which the unified report is constructed. We are suppose to care about the end result, no?